### PR TITLE
Update prometheus/prometheus to v0.51.1

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

supersedes https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31572

For https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31572#issuecomment-2029701231, the test case it is based on was removed upstream: https://github.com/prometheus/prometheus/commit/a28d7865adf6542cbac3b908124ae07846a0d4a8#diff-9978684151c256395cd22e4883a7e22c14e153a567a769c8abed86a2225a1514L462, so I think it is OK to change the expected outcome.  It allows trailing commas within brackets, which seems OK.

**Link to tracking Issue:**

**Testing:**

**Documentation:**

### CODE REVIEW

File: receiver/prometheusreceiver/testdata/openmetrics/bad_missing_or_extra_commas_2/metrics [hunk 1]
```diff
@@ -1,2 +0,0 @@
-a{a="1",b="2",} 1
```
Comment (songy23) [inline receiver/prometheusreceiver/testdata/openmetrics/bad_missing_or_extra_commas_2/metrics:1]: Could we keep this test but verify this case is working?

Comment (dashpole) [reply receiver/prometheusreceiver/testdata/openmetrics/bad_missing_or_extra_commas_2/metrics]: added back as a success vase.

**Comment by mx-psi** [comment]
Can you run `make genotelcontribcol && make genoteltestbedcol`?

**Comment by dashpole** [comment]
@mx-psi done

**Review by mx-psi — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Pull request merged**